### ISSUE TITLE
There was some error accessing the binary "keka7z"

### ISSUE THREAD

**Issue by Eakster** [aonez/Keka#208]
I used the old version Keka 1.08 and it works fine.
I got the new version Keka 1.10 and it gives me errors every time I extract any archives.
There was some error accessing the binary "keka7z" for ZIP
There was some error accessing the binary "kekaunrar" for RAR

I installed Keka into the Applications folder and gave it permissions to my hard drive
iMac Retina 5k, 27-inch 2017
macOS High Sierra 10.13.5
Fusion Drive (Mac OS Extended Journaled not APFS)
I tried to uninstall and reinstalled and remove all files with App Cleaner
I re downloaed
I set all the settings to normal
I still get the errors.

Any ideas?
Eakster

**Comment by aonez** [comment]
Could you test with this version and send the `Keka-log.log` that will appear in your Desktop?

https://t.keka.io/Keka-1.1.0r2717.zip

I think you're not alone, but I can't reproduce this issue anywhere so it's yet not clear why this happens.

Instead of using 1.0.8, use the latest (prior to 1.1.0) that is [1.0.16](https://github.com/aonez/Keka/releases/tag/v1.0.16).

**Comment by aonez** [comment]
Also, try those two. No log file needed here, just to see if they solve the issue:
- [Keka-1.1.0r2718.zip](https://github.com/aonez/Keka/files/2081807/Keka-1.1.0r2718.zip)
- [Keka-1.1.0r2719.zip](https://github.com/aonez/Keka/files/2081804/Keka-1.1.0r2719.zip)

**Comment by aonez** [comment]
@Eakster did it solved?

Here another version (from #213) that might have this fixed: [Keka-1.1.0r2729.zip](https://github.com/aonez/Keka/files/2093054/Keka-1.1.0r2729.zip)

**Comment by KeLeBoR** [comment]
I have the same mistake. The last working version is 1.1.0-rc.1...

**Comment by KeLeBoR** [comment]
Didn't fix the problem...

**Comment by aonez** [comment]
@KeLeBoR did you've tried the versions on this ticket?

- [Keka-1.1.0r2718](https://github.com/aonez/Keka/files/2081807/Keka-1.1.0r2718.zip)
- [Keka-1.1.0r2719](https://github.com/aonez/Keka/files/2081804/Keka-1.1.0r2719.zip)

Those versions create the file `Keka-log.log` in your desktop, please attach it here. Also, just in case, uninstall and reinstall Keka, downloading it again.

**Comment by KeLeBoR** [comment]
[Keka-log2718.log](https://github.com/aonez/Keka/files/2115807/Keka-log2718.log)
[Keka-log2719.log](https://github.com/aonez/Keka/files/2115808/Keka-log2719.log)

Hi annex,
r2718 is working, r2719 gave me the same error...
i hope the logs will help you to remedy the "bug".

**Comment by KeLeBoR** [comment]
@aonez yes i confirm r2718 work and r2719 didn't. Check it again. Try it in r2719 with the Terminal.app but nothing happen. i didn't have any special configuration in my machine (10.13.5 SSD APFS). and i also didn't have any trouble to give access to the disk. i try 1.1.1 again and install KekaDefaultApp v1.0.0-r.6 but it gave me again the error.

**Comment by aonez** [comment]
@KeLeBoR this one has the same attribute as r2718: [Keka-1.1.1r2742](https://github.com/aonez/Keka/files/2118214/Keka-1.1.1r2742.zip)

Works too? No need to check the Terminal. Thanks again for the feedback :)

**Comment by KeLeBoR** [comment]
@aonez Works without any problem...The only thing that i must is install the KekaDefaultApp v1.0.0-r.6 to set Keka as default compressor/uncompressor. I like to give feedback so that errors can also be corrected :-)

**Comment by robinalpha** [comment]
I have a similar problem. Keka was working perfectly fine until 1.1. Now everytime I try to decompress a file I have "L'extraction de _thefile_ a échoué. Code erreur 2 en utilisant "p7zip"
Erreur fatale"  (it's in french but I guess the translation is pretty obvious)

**Comment by aonez** [comment]
@robinalpha did you tried the revision 2742 from two comments above?

**Comment by robinalpha** [comment]
Yes and it didn't help. But actually I realized that I only saw it happens with some kind of zips. For example no problems to unzip the keka2747 zip but doesn't work with another one that the native mac app unzip without issues..

**Comment by aonez** [comment]
@robinalpha then this is another issue, probably just affecting that file. Can you share it?

@KeLeBoR I don't understand what you mean with:

> one mistake i must install the KekaDefaultApp v1.0.0-r.6

Sadly with sandbox Keka can't register as a default application itself, so it was getting rid of the functionality and button itself or finding a workaround, that was `KekaDefaultApp` extension. 😕

> I like to give feedback so that errors can also be corrected :-)

That is in fact what helps applications to grow 😃

**Comment by robinalpha** [comment]
It's seems indeed that it is affecting this file and some other not but i can also note that it happened a number of times with different files since 1.1, so its not just this file. But here it is.
[PUBLICACIÓN_PROYECTO_DE_RESOLUCIÓN_MODIFICACIÓN_MANUAL_DE_COMPENSACIONES_ (1).zip](https://github.com/aonez/Keka/files/2123719/PUBLICACION_PROYECTO_DE_RESOLUCION_MODIFICACION_MANUAL_DE_COMPENSACIONES_.1.zip)

**Comment by KeLeBoR** [comment]
@aonez Ah ok...Now i unstandand the context of KekaDefaultApp. Thanks for explain. :-)

**Comment by aonez** [comment]
@robinalpha the file linked extracted without a problem. Can you enable the [verbose output](https://github.com/aonez/Keka/wiki/Console-verbose) and let me know what it says?

**Comment by robinalpha** [comment]
I'm not sure, do you mean this?

par défaut	10:42:48.059218 -0500	launchservicesd	CHECKIN:0x0-0x70b70b 25339 com.aone.keka
par défaut	10:42:48.105264 -0500	NowPlayingTouchUI	[NowPlayingTouchUI] Reloading available apps due to workspace activating an application: NSConcreteNotification 0x600000256230 {name = NSWorkspaceDidActivateApplicationNotification; object = <NSWorkspace: 0x604000001560>; userInfo = {
    NSWorkspaceApplicationKey = "<NSRunningApplication: 0x6000003029a0 (com.aone.keka - 25339)>";
}}
par défaut	10:42:48.197550 -0500	Keka	Updating user defaults from 2742...
par défaut	10:42:48.203852 -0500	Keka	Dev mode enabled
par défaut	10:42:48.205625 -0500	Keka	Running on OS X/macOS version 10.13.5 (assemblage 17F77)
par défaut	10:42:48.206007 -0500	Keka	Running Keka (WEB) version 1.1.1 (2742)
par défaut	10:42:48.207500 -0500	Keka	Used localization: fr
par défaut	10:42:48.458141 -0500	Keka	Adding 1 files
par défaut	10:42:48.458640 -0500	Keka	Waiting for more input: 0.250000
par défaut	10:42:48.459545 -0500	Keka	Requesting access to all bookmarks...
par défaut	10:42:48.497480 -0500	Keka	DEPRECATION: SUUpdater is now deprecated. Please use SPUStandardUpdaterController as a nib instantiated replacement, or SPUUpdater.
par défaut	10:42:48.820959 -0500	Keka	Format detected: Zip
par défaut	10:42:48.822065 -0500	Keka	Binary used: keka7z
par défaut	10:42:48.822433 -0500	Keka	Arguments: (
    x,
    "/Volumes/DATA/Downloads/PUBLICACIO\U0301N_PROYECTO_DE_RESOLUCIO\U0301N_MODIFICACIO\U0301N_MANUAL_DE_COMPENSACIONES_ (1).zip",
    "-o/Volumes/DATA/Downloads/.Keka-058BCCEE-960F-4346-BB41-4F5E50032230",
    "-aou",
    "-xr!__MACOSX"
)
par défaut	10:42:49.325482 -0500	Keka	done: 1
par défaut	10:42:58.120158 -0500	Keka	LSExceptions shared instance invalidated for timeout.

there's still the same error showing up and the zip isn't decompressed

**Comment by aonez** [comment]
@robinalpha that's it thanks! It's strange because even in the logs it does not appear to be any problem...

**Comment by robinalpha** [comment]
and at the same time.. 
<img width="399" alt="screen shot 2018-06-22 10 52 23 am" src="https://user-images.githubusercontent.com/15682080/41786250-a7cfbef0-760a-11e8-84d6-c1faf80958fc.png">

**Comment by aonez** [comment]
Where are you trying to extract the file? Can you try in the Desktop?

**Comment by robinalpha** [comment]
I'm doing it in my download folder. trying on the desktop didn't change anything.

**Comment by aonez** [comment]
@robinalpha what format is in the "DATA" disk?

**Comment by robinalpha** [comment]
APFS encrypted

**Comment by aonez** [comment]
@robinalpha please activate the reader log, that will help to better understand the issue:

```sh
defaults write com.aone.keka DevLogReader -bool <TRUE/FALSE>
```

Hope we can find a solution for this problem :)

**Comment by robinalpha** [comment]
I'm not sure where I'm supposed to read the log.. is it in the console? like this..?

par défaut	10:29:15.393403 -0500	launchservicesd	CHECKIN:0x0-0xe99e99 53459 com.aone.keka
par défaut	10:29:15.426472 -0500	NowPlayingTouchUI	[NowPlayingTouchUI] Reloading available apps due to workspace activating an application: NSConcreteNotification 0x6000004463c0 {name = NSWorkspaceDidActivateApplicationNotification; object = <NSWorkspace: 0x604000001560>; userInfo = {
    NSWorkspaceApplicationKey = "<NSRunningApplication: 0x6000003014d0 (com.aone.keka - 53459)>";
}}
par défaut	10:29:15.480404 -0500	Keka	Updating user defaults from 2742...
par défaut	10:29:15.932337 -0500	Keka	DEPRECATION: SUUpdater is now deprecated. Please use SPUStandardUpdaterController as a nib instantiated replacement, or SPUUpdater.
par défaut	10:29:16.149685 -0500	org.sparkle-project.Downloader	Faulting in NSHTTPCookieStorage singleton
par défaut	10:29:16.149758 -0500	org.sparkle-project.Downloader	Faulting in CFHTTPCookieStorage singleton
par défaut	10:29:16.150128 -0500	org.sparkle-project.Downloader	Creating default cookie storage with default identifier
par défaut	10:29:16.161822 -0500	org.sparkle-project.Downloader	TIC Enabling TLS [1:0x7f88a8f03bf0]
par défaut	10:29:16.161875 -0500	org.sparkle-project.Downloader	TIC TCP Conn Start [1:0x7f88a8f03bf0]
par défaut	10:29:16.162063 -0500	org.sparkle-project.Downloader	Task <66155F37-BC29-4E07-A584-3207E5D6C64A>.<1> setting up Connection 1
par défaut	10:29:16.162098 -0500	org.sparkle-project.Downloader	[1 <private> <private>] start
par défaut	10:29:16.655894 -0500	org.sparkle-project.Downloader	TIC TLS Event [1:0x7f88a8f03bf0]: 1, Pending(0)
par défaut	10:29:16.846657 -0500	org.sparkle-project.Downloader	TIC TLS Event [1:0x7f88a8f03bf0]: 2, Pending(0)
par défaut	10:29:16.909936 -0500	org.sparkle-project.Downloader	TIC TLS Event [1:0x7f88a8f03bf0]: 11, Pending(0)
par défaut	10:29:16.911551 -0500	org.sparkle-project.Downloader	TIC TLS Event [1:0x7f88a8f03bf0]: 12, Pending(0)
par défaut	10:29:16.913245 -0500	org.sparkle-project.Downloader	TIC TLS Event [1:0x7f88a8f03bf0]: 14, Pending(0)
par défaut	10:29:17.290899 -0500	org.sparkle-project.Downloader	System Trust Evaluation yielded status(0)
par défaut	10:29:17.303458 -0500	org.sparkle-project.Downloader	TIC TLS Trust Result [1:0x7f88a8f03bf0]: 0
par défaut	10:29:17.492520 -0500	org.sparkle-project.Downloader	TIC TLS Event [1:0x7f88a8f03bf0]: 20, Pending(0)
par défaut	10:29:17.492852 -0500	org.sparkle-project.Downloader	TIC TCP Conn Connected [1:0x7f88a8f03bf0]: Err(16)
par défaut	10:29:17.493257 -0500	org.sparkle-project.Downloader	TIC TCP Conn Event [1:0x7f88a8f03bf0]: 1
par défaut	10:29:17.493453 -0500	org.sparkle-project.Downloader	TIC TCP Conn Event [1:0x7f88a8f03bf0]: 8
par défaut	10:29:17.493495 -0500	org.sparkle-project.Downloader	TIC TLS Handshake Complete [1:0x7f88a8f03bf0]
par défaut	10:29:17.506702 -0500	org.sparkle-project.Downloader	Task <66155F37-BC29-4E07-A584-3207E5D6C64A>.<1> now using Connection 1
par défaut	10:29:17.506816 -0500	org.sparkle-project.Downloader	Task <66155F37-BC29-4E07-A584-3207E5D6C64A>.<1> sent request, body N
par défaut	10:29:17.718838 -0500	org.sparkle-project.Downloader	Task <66155F37-BC29-4E07-A584-3207E5D6C64A>.<1> received response, status 301 content K
par défaut	10:29:17.720162 -0500	org.sparkle-project.Downloader	Task <66155F37-BC29-4E07-A584-3207E5D6C64A>.<1> response ended
par défaut	10:29:17.721015 -0500	org.sparkle-project.Downloader	Task <66155F37-BC29-4E07-A584-3207E5D6C64A>.<1> now using Connection 1
par défaut	10:29:17.721158 -0500	org.sparkle-project.Downloader	Task <66155F37-BC29-4E07-A584-3207E5D6C64A>.<1> sent request, body N
par défaut	10:29:17.950647 -0500	org.sparkle-project.Downloader	Task <66155F37-BC29-4E07-A584-3207E5D6C64A>.<1> received response, status 302 content K
par défaut	10:29:17.951426 -0500	org.sparkle-project.Downloader	Task <66155F37-BC29-4E07-A584-3207E5D6C64A>.<1> response ended
par défaut	10:29:17.952317 -0500	org.sparkle-project.Downloader	Task <66155F37-BC29-4E07-A584-3207E5D6C64A>.<1> now using Connection 1
par défaut	10:29:17.952475 -0500	org.sparkle-project.Downloader	Task <66155F37-BC29-4E07-A584-3207E5D6C64A>.<1> sent request, body N
par défaut	10:29:18.153513 -0500	org.sparkle-project.Downloader	Task <66155F37-BC29-4E07-A584-3207E5D6C64A>.<1> received response, status 200 content K
par défaut	10:29:18.161376 -0500	org.sparkle-project.Downloader	Task <66155F37-BC29-4E07-A584-3207E5D6C64A>.<1> response ended
par défaut	10:29:18.176976 -0500	org.sparkle-project.Downloader	TIC TCP Conn Cancel [1:0x7f88a8f03bf0]
par défaut	10:29:18.177531 -0500	org.sparkle-project.Downloader	[1 <private> stream, pid: 53461, url: https://www.keka.io/update, traffic class: 200, tls] cancelled
	[1.1 D9F58E82-4000-4853-97AC-C05946175734 <private>.56336<-><private>]
	Connected Path: satisfied (Path is satisfied), interface: en0, ipv4, dns
	Duration: 2.015s, DNS @0.007s took 0.142s, TCP @0.154s took 0.334s, TLS took 0.841s
	bytes in/out: 14378/811, packets in/out: 17/10, rtt: 0.185s, retransmitted packets: 0, out-of-order packets: 0
par défaut	10:29:18.178242 -0500	org.sparkle-project.Downloader	TIC TCP Conn Destroyed [1:0x7f88a8f03bf0]
erreur	10:29:18.179238 -0500	org.sparkle-project.Downloader	TIC Read Status [1:0x0]: 1:57
erreur	10:29:18.179299 -0500	org.sparkle-project.Downloader	TIC Read Status [1:0x0]: 1:57
par défaut	10:29:25.464826 -0500	Keka	LSExceptions shared instance invalidated for timeout.

**Comment by aonez** [comment]
@robinalpha yes that will be, but try to extract the file that's giving you the issue so it shows in the log. Thanks :)

**Comment by robinalpha** [comment]
That’s what I did.. I tried again, twice, and here’s the result. But it doesn’t seem much, so maybe it’s not the right way to get the log..?

par défaut	10:46:36.451934 -0500	launchservicesd	CHECKIN:0x0-0xea6ea6 53731 com.aone.keka
par défaut	10:46:36.474545 -0500	NowPlayingTouchUI	[NowPlayingTouchUI] Reloading available apps due to workspace activating an application: NSConcreteNotification 0x600000640bd0 {name = NSWorkspaceDidActivateApplicationNotification; object = <NSWorkspace: 0x604000001560>; userInfo = {
    NSWorkspaceApplicationKey = "<NSRunningApplication: 0x60000030d890 (com.aone.keka - 53731)>";
}}
par défaut	10:46:36.563311 -0500	Keka	Updating user defaults from 2742...
par défaut	10:46:36.952452 -0500	Keka	DEPRECATION: SUUpdater is now deprecated. Please use SPUStandardUpdaterController as a nib instantiated replacement, or SPUUpdater.
par défaut	10:46:46.567401 -0500	Keka	LSExceptions shared instance invalidated for timeout.
par défaut	10:46:54.832746 -0500	NowPlayingTouchUI	[NowPlayingTouchUI] Reloading available apps due to workspace activating an application: NSConcreteNotification 0x600000456da0 {name = NSWorkspaceDidActivateApplicationNotification; object = <NSWorkspace: 0x604000001560>; userInfo = {
    NSWorkspaceApplicationKey = "<NSRunningApplication: 0x60000030e070 (com.aone.keka - 53731)>";
}}
par défaut	10:47:04.075989 -0500	launchservicesd	CHECKIN:0x0-0xea8ea8 53744 com.aone.keka
par défaut	10:47:04.098762 -0500	NowPlayingTouchUI	[NowPlayingTouchUI] Reloading available apps due to workspace activating an application: NSConcreteNotification 0x600000442f70 {name = NSWorkspaceDidActivateApplicationNotification; object = <NSWorkspace: 0x604000001560>; userInfo = {
    NSWorkspaceApplicationKey = "<NSRunningApplication: 0x60000030d890 (com.aone.keka - 53744)>";
}}
par défaut	10:47:04.129926 -0500	Keka	Updating user defaults from 2742...
par défaut	10:47:04.451506 -0500	Keka	DEPRECATION: SUUpdater is now deprecated. Please use SPUStandardUpdaterController as a nib instantiated replacement, or SPUUpdater.
par défaut	10:47:14.196672 -0500	Keka	LSExceptions shared instance invalidated for timeout.

**Comment by aonez** [comment]
@robinalpha I'll prepare another build that will post all logs on the desktop file, hopefully this will help :)

**Comment by aonez** [comment]
@robinalpha can you try with the newly released 1.1.2 version?

**Comment by aonez** [comment]
@KeLeBoR an can you also try with the newly released 1.1.2 version?

**Comment by robinalpha** [comment]
@aonez seems like it works! :)

**Comment by aonez** [comment]
Thanks @KeLeBoR and @robinalpha!

**Comment by crax** [comment]
I'm using the 1.1.2(2787) version, still got the same error. @aonez

**Comment by gingerbeardman** [comment]
I wonder if the app (or just kekaunrar) is suffering from App Translocation?

**Comment by aonez** [comment]
> Strangely both report themselves as 1.1.2 (2787) the only difference is that the one from the current DMG is hfs+ compressed (20MB) vs my backup (28MB)?

@gingerbeardman Maybe the older version from the backup was silently updated using Sparkle? That way uses the ZIP package so it's not HFS+ compressed. You can reproduce it with the DMG package? And with the ZIP one: [Keka-1.1.2.zip](https://github.com/aonez/Keka/releases/download/v1.1.2/Keka-1.1.2.zip)?

@crax are you using the web version? If you're, please try with the ZIP packed version: [Keka-1.1.2.zip](https://github.com/aonez/Keka/releases/download/v1.1.2/Keka-1.1.2.zip). Let me know if you're using the MAS version.

**Comment by aonez** [comment]
> I wonder if the app (or just kekaunrar) is suffering from App Translocation?

@gingerbeardman this was the first thing I've checked when I saw you had this issue too... The problem is I'm unable to reproduce it :(

**Comment by GregOriol** [comment]
Having this issue with a freshly downloaded 1.1.5 web vesion on MacOS Mojave, APFS
Any update on the issue? how could I help investigate?

**Comment by GregOriol** [comment]
I actually have something interesting: message in the console
`exec of /Applications/Keka.app/Contents/Resources/keka7z denied since it was quarantined by Safari and not approved by Gatekeeper, qtn-flags was 0x00000083`

**Comment by aonez** [comment]
Wow thanks @GregOriol! So you've downloaded the ZIP package instead of the DMG, and extracted it with Keka? Or Safari extracted it automatically?

**Comment by GregOriol** [comment]
@aonez I tried both dmg from the website and zip from github releases, both using Safari. No automatic extract, I opened them myself. Both failed.

**Comment by aonez** [comment]
@GregOriol can you print this from the DMG/ZIP:

```sh
xattr -l ~/Downloads/Keka-1.1.5.dmg > ~/Desktop/dmg-xattr.txt
```

And this for the app itself:

```sh
xattr -l -r /Applications/Keka.app > ~/Desktop/app-xattr.txt
```

I'm assuming you have the DMG/ZIP package in your Downloads folder, change it accordingly. ZIP the resulting `dmg-xattr.txt` and `app-xattr.txt` and attach the ZIP here.

By the way version 1.1.6 was just released, you can check if this new download still fails for you. But will be interesting if you can keep 1.1.5 version, at least for test purposed. Thanks!!!

**Comment by aonez** [comment]
@GregOriol Also, not sure it will keep the quarantine, but if you can compress the `Keka.app` is failing in a TAR.GZ, then compress tar tarball in ZIP and attach it here.

**Comment by GregOriol** [comment]
@aonez As I had a working version of Keka after the manual download yesterday, I downloaded the dmg from the website again using Safari to perform the tests. So it's 1.1.6 and it's failing the same way. Here are the results of xattr:
[keka-dmg-xattr.txt](https://github.com/aonez/Keka/files/2615315/keka-dmg-xattr.txt)
[keka-app-xattr.txt](https://github.com/aonez/Keka/files/2615314/keka-app-xattr.txt)

Making a tgz of the non-working Keka.app (cvzf) and extracting it using tar (xvzf) seems to preserve the quarantine. But extracting it using Keka doesn't seem to preserve the quarantine. Here is the tgz/zip:
[Keka.tgz.zip](https://github.com/aonez/Keka/files/2615363/Keka.tgz.zip)

I also tried downloading it from the website using Firefox, and it works fine. I'm attaching xattr results from it too if that might help:
[keka-app-firefox-xattr.txt](https://github.com/aonez/Keka/files/2615374/keka-app-firefox-xattr.txt)
[keka-dmg-firefox-xattr.txt](https://github.com/aonez/Keka/files/2615375/keka-dmg-firefox-xattr.txt)

I also tried this command to remove the quarantine and it works:
```
find Keka.app -name '*' -print0 | xargs -0 xattr -d com.apple.quarantine
```

**Comment by aonez** [comment]
Thanks a lot @GregOriol! I'll check the files.

> find Keka.app -name '*' -print0 | xargs -0 xattr -d com.apple.quarantine

This should work too, then:

```sh
xattr -d -r com.apple.quarantine /Applications/Keka.app
```

**Comment by GregOriol** [comment]
I confirm your command works too.

**Comment by aonez** [comment]
@GregOriol the extracted `Keka.tgz.zip` app retains the quarantine but does not fail here... 🤯

**Comment by GregOriol** [comment]
@aonez Sorry can't help here, I don't know how this behavior is working :-/ Maybe there are some ways to disable it and you have it set on your macOS? or maybe I/we do have something set on our macs?..

**Comment by aonez** [comment]
@GregOriol yeah really don't know either. Restarting the Mac does not cause any difference, does it?

At least now with your tests and feedback we have a cause and a workaround 🎉

**Comment by gingerbeardman** [comment]
I think the problem is a combination of Quarantine and App Translocation.
